packages/utilities/src/useFocusRects.ts

```diff
@@ -78,19 +129,26 @@ export const FocusRects: React.FunctionComponent<{ rootRef?: React.RefObject<HTM
   return null;
 };
 
-function _onMouseDown(ev: MouseEvent): void {
-  setFocusVisibility(false, ev.target as Element);
+function _onMouseDown(ev: MouseEvent, providerElem?: Element): void {
+  setFocusVisibility(false, ev.target as Element, providerElem);
 }
 
-function _onPointerDown(ev: PointerEvent): void {
+function _onPointerDown(ev: PointerEvent, providerElem?: Element): void {
   if (ev.pointerType !== 'mouse') {
-    setFocusVisibility(false, ev.target as Element);
+    setFocusVisibility(false, ev.target as Element, providerElem);
+  }
+}
+
+function _onKeyDown(ev: KeyboardEvent, providerElem?: Element): void {
+  // eslint-disable-next-line deprecation/deprecation
+  if (isDirectionalKeyCode(ev.which)) {
+    setFocusVisibility(true, ev.target as Element, providerElem);
   }
 }
 
-function _onKeyDown(ev: KeyboardEvent): void {
+function _onKeyUp(ev: KeyboardEvent, providerElem?: Element): void {
   // eslint-disable-next-line deprecation/deprecation
   if (isDirectionalKeyCode(ev.which)) {
-    setFocusVisibility(true, ev.target as Element);
+    setFocusVisibility(true, ev.target as Element, providerElem);
   }
 }
```

Comment:
Maybe add a comment about why you need to listen for both onKeyDown and onKeyUp (and why both functions do the same thing)?

Reply:
This is a good suggestion, just added it.